js/components/search.js

```diff
@@ -5,85 +5,165 @@ $( document ).ready(function() {
-    $.ajax({
-      url: url,
-      cache: false,
-      dataType: 'json'
-    })
-      .done(function(json) {
-        $('.loading').remove();
-        $('.search-results-count').append( json.hits.total + ' search results');
-        if (json.hits.total == 0){
-          $('.search-no-results').show();
-        }
-        else{
-          $('.search-no-results').remove();
-        }
-
-        $.each(json.hits.hits, function(i, hit){
-
-        var result_description = hit._source.description;
-
-          var tags = '';
-          if(hit._source.tag) {
-
-            $.each(hit._source.tag, function(i, tag) {
-              tags += '<span class="search-result-list-tag">&nbsp;<a href="../search-results/?q='
-              + tag + '" title="Search for '
-              + tag +'">'
-              + tag + '</a>&nbsp;/</span>';
-            });
-          }
-
-          var content_type = '';
-          if(hit._source.content_type == 'text/html') {
-            content_type = '<span class="glyphicon glyphicon-link"></span> Website';
-          }
-          else if(hit._source.content_type == 'application/pdf') {
-            content_type = '<span class="glyphicon glyphicon-file"></span> PDF';
-          }
-          else {
-            content_type = '<span class="glyphicon glyphicon-file"></span> ' + hit._source.content_type + '';
-          }
-          $('.search-results-container').append('<article class="search-result-list"><h1><a href="'
-            + hit._source.url
-            + '" target="_blank">'
-            + hit._source.title
-            + '</a></h1>'
-            + '<p>'
-            + result_description
-            + '</p>'
-            +'<p>Tagged&nbsp;/'
-            + tags
-            + '</p>'
-            + '</article>');
-        });
-      });
+  /**
+   * Returns URL for search request
+   *
+   * @param {string} query - Raw string holding search terms/query
+   * @param {string} apiKey - Raw string holding API key for api.data.gov
+   * @param {object} [options] - Optional search parameters
+   * @param {number} [options.size=10] - How many results to fetch
+   * @param {number} [options.from=0] - Offset to start from for paging
+   * @returns {string} URL for desired search
+   */
+  function apiSearchUrl( query, apiKey, options ) {
+    var q = encodeURIComponent( query );
+    var size = (options && options.size) || 10;
+    var from = (options && options.from) || 0;
+
+    return (
+      'https://api.data.gov/beckley-federalist/v0/resources/eiti/?' +
+      'q=' + q + '&size=' + size + '&from=' + from + '&api_key=' + apiKey
+    );
+  }
+
+  function localSearchUrl( query, options ) {
+    var q = encodeURIComponent( query );
+    var size = (options && options.size) || 10;
+    var from = (options && options.from) || 0;
+
+    return '/search-results/?q=' + q + '&size=' + size + '&from=' + from;
```

Comment:
The original code had `../search-results/` and I wondered why the `../` was there instead of using an absolute reference. Obviously, I took it out but could happily put it back.